### ISSUE TITLE
Build failed:  use of undeclared identifier 'isTypedArray'; did you mean 'expo::isTypedArray'?

### ISSUE THREAD

**Issue by baylesa-dev** [mrousavy/react-native-mmkv#465]
Hi there!

While building my react-native app with `expo@46` (for`expo-modules`), compilation fails:

`package.json`:
```json
  "dependencies": {
    "expo": "^46.0.0",
    "react": "18.1.0",
    "react-native": "0.70.3",
    "react-native-mmkv": "^2.5.0"
  },
```

```
Compiling MmkvHostObject.mm

❌  node_modules/react-native-mmkv/ios/MmkvHostObject.mm:89:13: use of undeclared identifier 'isTypedArray'; did you mean 'expo::isTypedArray'?
        if (isTypedArray(runtime, object)) {
          ^~~~
❌  node_modules/react-native-mmkv/ios/MmkvHostObject.mm:91:29: use of undeclared identifier 'getTypedArray'
          auto typedArray = getTypedArray(runtime, object);
     ^
❌  node_modules/react-native-mmkv/ios/MmkvHostObject.mm:196:20: use of undeclared identifier 'TypedArrayKind'; did you mean 'expo::TypedArrayKind'?
        TypedArray<TypedArrayKind::Uint8Array> array(runtime, data.length);
                            ^
❌  node_modules/react-native-mmkv/ios/MmkvHostObject.mm:196:47: expected ';' after expression
        TypedArray<TypedArrayKind::Uint8Array> array(runtime, data.length);
           ^
❌  node_modules/react-native-mmkv/ios/MmkvHostObject.mm:196:9: use of undeclared identifier 'TypedArray'
        TypedArray<TypedArrayKind::Uint8Array> array(runtime, data.length);
                                              ^
❌  node_modules/react-native-mmkv/ios/MmkvHostObject.mm:196:48: use of undeclared identifier 'array'
        TypedArray<TypedArrayKind::Uint8Array> array(runtime, data.length);
        ^
❌  node_modules/react-native-mmkv/ios/MmkvHostObject.mm:200:9: use of undeclared identifier 'array'; did you mean 'std::array'?
        array.update(runtime, vector);
                                               ^
❌  node_modules/react-native-mmkv/ios/MmkvHostObject.mm:200:9: use of undeclared identifier 'array'; did you mean 'std::array'?
        array.update(runtime, vector);
                                                               ^
❌  node_modules/react-native-mmkv/ios/MmkvHostObject.mm:200:9: unknown type name 'array'
        array.update(runtime, vector);
                                                               ^
❌  node_modules/react-native-mmkv/ios/MmkvHostObject.mm:200:14: cannot use dot operator on a type
        array.update(runtime, vector);
        ^
❌  node_modules/react-native-mmkv/ios/MmkvHostObject.mm:201:16: use of undeclared identifier 'array'
        return array;
```

Does anyone have an idea how to fix this?

**Comment by bombillazo** [comment]
Hello, having the same issue as well.

**Comment by bombillazo** [comment]
I think https://github.com/mrousavy/react-native-mmkv/releases/tag/v2.5.0 introduced a regression, which is crazy cause we haven't even updated our `react-native-mmkv` package version to 2.5.0 but it is the commit I see that introduced `isTypedArray`

**Comment by bombillazo** [comment]
@baylesa-dev our solution for the moment is to change `"react-native-mmkv": "^2.4.4",` to "react-native-mmkv": "2.4.4"

**Comment by TowhidKashem** [comment]
hey @mrousavy , just a heads up the current version is breaking builds, I'm using the above suggestion to pin to version `2.4.4` as a temp workaround.

**Comment by mrousavy** [comment]
Fixed in 740c81065af2ffb1a6d6132d922d732bc52e4c1e, released in 2.5.1 ✅
